cli/src/movement_util.rs

```diff
@@ -109,16 +109,20 @@ impl Direction {
         args: &MovementArgsInternal,
     ) -> Result<Rc<RevsetExpression>, CommandError> {
         let target_revset = match self {
-            Direction::Next => if args.conflict {
-                start_revset
-                    .children()
-                    .descendants()
-                    .filtered(RevsetFilterPredicate::HasConflict)
-                    .roots()
-            } else {
-                start_revset.descendants_at(args.offset)
+            Direction::Next => {
+                if args.conflict {
+                    start_revset
+                        .children()
+                        .minus(working_revset)
+                        .descendants()
+                        .filtered(RevsetFilterPredicate::HasConflict)
+                        .roots()
+                } else {
+                    start_revset
+                        .descendants_at(args.offset)
+                        .minus(working_revset)
+                }
```

Comment:
BTW, I think the query can be reorganized as
```rust
let next = if !args.edit {
    start.children().minus(wc)
} else {
    start.children()
};
if args.conflict {
    next.descendants().filtered(HasConflict).roots()
} else {
    next.descendants_at(N - 1)
}
```

Reply:
Thanks, this was helpful. I'm finally getting used to revsets!

I've folded this in, though for consistency, I've used to match expressions so that it's clearer how `Direction::Prev` is treated. 

Using only one match expressions ends up with too much duplicated code that is just so slightly different, but a single match seems just right.